### ISSUE TITLE
Custom filebeat module timezone conversion issue

### ISSUE THREAD

**Issue by nanjum88** [elastic/beats#18697]
We build a custom module for parsing F5 Load Balancer logs, all the patterns are working fine. But the timezone is set to UTC -7 hours for the event ingested and showing up in Kibana (Browser based timezone). We're currently in PT timezone.

Issue reported on elastic discussion here - (https://discuss.elastic.co/t/custom-filebeat-module-timezone-conversion-issue/233428)

Following is the code for processing the event.

`
     {
        "grok" : {
          "field" : "message",
          "patterns" : [
            "%{SYSLOGTIMESTAMP:f5.timestamp} %{IPORHOST:host.hostname} (%{LOGLEVEL:log.level})? %{PROG:process.parent.name}\\(%{NOTSPACE:process.name}\\)(?:\\[%{POSINT:process.pid:long}\\])?: %{GREEDYDATA:syslog_message}",
            "%{SYSLOGTIMESTAMP:f5.timestamp} %{IPORHOST:host.hostname} %{PROG:process.name}(?:\\[%{POSINT:process.pid:long}\\])?: %{GREEDYDATA:syslog_message}"
          ],
          "ignore_missing" : true
        }
      },
{
        "date" : {
        "if": "ctx.event?.timezone == null",           
          "field" : "f5.timestamp",
          "target_field" : "@timestamp",
          "formats" : [
            "MMM  d HH:mm:ss",
            "MMM dd HH:mm:ss"
          ]
        }
      },
       {
        "date" : {
          "if": "ctx.event?.timezone != null",           
          "field" : "f5.timestamp",
          "timezone": "{{ event.timezone }}",           
          "target_field" : "@timestamp",
          "formats" : [
            "MMM  d HH:mm:ss",
            "MMM dd HH:mm:ss"
          ]
        }
      },       
      {
        "date_index_name" : {
          "field" : "@timestamp",
          "date_rounding" : "d",
          "index_name_prefix" : "f5-",
          "index_name_format" : "yyyy.MM.dd",
          "ignore_failure" : false
        }
      }
`

Sample Events - 
`
May 13 11:01:03 sysloghost notice run-parts(/etc/cron.hourly)[12939]: finished genkeys
May 19 23:31:27 sysloghost tmm3[12488]: Rule /Common/geo-blocking-irule <CLIENT_ACCEPTED>: Dropping request from GEO. VS IP: 10.0.0.0, Client IP: 10.0.1.1
`
Filebeat version - 7.5.1

I believe the date processor is not converting the date obtained from syslog into UTC. As in the following example event after ingest, I would expect @timestamp 2020-05-17T20:24:43.000Z instead of 2020-05-17T13:24:43.000Z.

`{
  "_index": "f5-2020.05.17",
  "_type": "_doc",
  "_id": "RJ7EMHIB-5un80h6CouL",
  "_version": 1,
  "_score": null,
  "_source": {
    "agent": {
      "hostname": "syslog",
      "id": "f9dbf01a-0100-47e9-b799-5e7c28f3cf73",
      "type": "filebeat",
      "ephemeral_id": "3a679547-3250-4cfd-9163-acbaf07026d7",
      "version": "7.5.1"
    },
    "process": {
      "name": "ssl_acc"
    },
    "log": {
      "file": {
        "path": "/opt/syslog-ng/f5apm/lb02.corp.sac/2020051713.log"
      },
      "offset": 40555,
      "level": "info"
    },
    "message": "May 17 13:24:43 sysloghost info logger: [ssl_acc] 10.1.1.1 - admin [17/May/2020:13:24:43 -0700] \"/mgmt/shared/inflate/available\" 200 2",
    "fileset": {
      "name": "logs"
    },
    "tags": [
      "infosec",
      "network",
      "firewall",
      "pan"
    ],
    "f5": {},
    "input": {
      "type": "log"
    },
    "@timestamp": "2020-05-17T13:24:43.000Z",
    "ecs": {
      "version": "1.1.0"
    },
    "service": {
      "type": "f5"
    },
    "host": {
      "hostname": "sysloghost",
      "os": {
        "kernel": "2.6.32-754.25.1.el6.x86_64",
        "codename": "Santiago",
        "name": "Red",
        "family": "redhat",
        "version": "6.10 (Santiago)",
        "platform": "redhat"
      },
      "containerized": false,
      "name": "syslog",
      "id": "a8c1843410a5e496c9dd618500000011",
      "architecture": "x86_64"
    },
    "event": {
      "module": "f5",
      "dataset": "f5.logs"
    },
    "fields": {
      "environment": "prod",
      "project": "pan_logs"
    }
  },
  "fields": {
    "@timestamp": [
      "2020-05-17T13:24:43.000Z"
    ]
  },
  "sort": [
    1589721883000
  ]
}`

Followed some old threads from filebeat Forums and even tried to delete the pipeline from dev tools and then re-create the pipeline, but still the events show-up in Kibana with wrong timezone - https://discuss.elastic.co/t/filebeat-system-module-auth-log-timezone-conversion-problem/140192/6

![image](https://user-images.githubusercontent.com/32209105/82604604-66fdc580-9b69-11ea-8d03-f270139bb0d6.png)

**Comment by tzeappa** [comment]
Since upgrading to Filebeat 7.7.0 it seems that the `@timestamp` field is no longer UTC for `type: log` and just basic log parsing without any modules and it just parses the OS date without any timezone conversion to UTC (as it should do and it does for example for the Apache module).
I have made sure that the `event.timezone` field is added properly and I have set `var.convert_timezone: true` in filebeat but nothing seems to help.

This looks like a regression bug that should be fixed as I have found articles describing older version doing this but usually it was resolved with `var.convert_timezone: true`.
I've also opened this topic https://discuss.elastic.co/t/filebeat-7-7-0-timestamp-not-utc/235396

Hopefully, a screenshot should explain better how the UTC+2 is registered as UTC and another 2 hours are added when showing data.
![Screen Shot 2020-06-04 at 14 12 56](https://user-images.githubusercontent.com/2735143/83750125-7e5ea780-a66d-11ea-90fe-eb06e46ed375.png)
